File: 2007-find-original-array-from-doubled-array/2007-find-original-array-from-doubled-array.java
class Solution {
    public int[] findOriginalArray(int[] changed) {
        //checking edge condition of odd and zero length
        if(changed.length%2 != 0 || changed.length == 0)
            return new int[0];
        
        List<Integer> ans = new ArrayList();
        // Sorting the array
        Arrays.sort(changed);
        Queue<Integer> q = new LinkedList<Integer>();
        
        for(int i = 0; i < changed.length; i++)
        {
            // if queue is not empty and front of the queue is same as in changed array then pop/remove it from queue 
            if(!q.isEmpty() && changed[i] == q.peek()){
                q.poll();
            }
            else{// add the double of current element to the queue 
                q.add(changed[i]*2);
                ans.add(changed[i]); // add current element to answer array
                //idx++;
            }
        }
        // (ans.stream().mapToInt(i->i).toArray();) changing Integet to int in array
        return !q.isEmpty()?new int[0]: ans.stream().mapToInt(i->i).toArray();
    }
}

File: 2007-find-original-array-from-doubled-array/2007-find-original-array-from-doubled-array.cpp
class Solution {
public:
    vector<int> findOriginalArray(vector<int>& changed) {
        if(changed.size()%2 != 0)
            return vector<int>{};
        vector<int> ans;
        sort(changed.begin(),changed.end());
        queue<int>q;
        
        for(int i = 0; i < changed.size(); i++)
        {
            if(!q.empty() && changed[i] == q.front()){
                q.pop();
            }
            else{
                q.push(changed[i]*2);
                ans.push_back(changed[i]);
            }
        }
        
        return !q.empty()?vector<int>{}: ans;
    }
};